File: solutions/elixir/captains-log/lib/captains_log.ex
defmodule CaptainsLog do
  @planetary_classes ["D", "H", "J", "K", "L", "M", "N", "R", "T", "Y"]

  def random_planet_class(), do: Enum.random(@planetary_classes)

  def random_ship_registry_number() do
    "NCC-#{:rand.uniform(8999) + 1000}"
  end

  def random_stardate() do
    :rand.uniform() * 1000 + 41000
  end

  def format_stardate(stardate) do
    :io_lib.format("~.1f", [stardate]) |> to_string
  end
end

File: solutions/julia/captains-log/captains-log.jl
using Random

function random_planet()
    rand(['D', 'H', 'J', 'K', 'L', 'M', 'N', 'R', 'T', 'Y'])
end

function random_ship_registry_number()
    "NCC-$(rand(1000:9999))"
end

function random_stardate()
    1000. * rand() + 41000.
end

function random_stardate_v2()
    rand(41000.0:0.1:42000.0)
end

function pick_starships(starships, number_needed)
    Random.shuffle(starships)[1:number_needed]
end
